pkg/registry/thirdpartyresource/etcd/etcd_test.go

```diff
@@ -80,26 +80,24 @@ func TestCreate(t *testing.T) {
 func TestUpdate(t *testing.T) {
 	storage, fakeClient := newStorage(t)
 	test := resttest.New(t, storage, fakeClient.SetError)
-	key, err := storage.KeyFunc(test.TestContext(), "foo")
-	if err != nil {
-		t.Fatal(err)
-	}
-	key = etcdtest.AddPrefix(key)
-	fakeClient.ExpectNotFoundGet(key)
-	fakeClient.ChangeIndex = 2
-	rsrc := validNewThirdPartyResource("foo")
-	existing := validNewThirdPartyResource("exists")
-	existing.Namespace = test.TestNamespace()
-	obj, err := storage.Create(test.TestContext(), existing)
-	if err != nil {
-		t.Fatalf("unable to create object: %v", err)
-	}
-	older := obj.(*expapi.ThirdPartyResource)
-	older.ResourceVersion = "1"
 	test.TestUpdate(
-		rsrc,
-		existing,
-		older,
+		// valid
+		validNewThirdPartyResource("foo"),
+		func(ctx api.Context, obj runtime.Object) error {
```

Comment:
Actually, why do we even need registrytest?
What if resttest directly uses storage.(rest.Getter).Get instead of this getFn and similarly storage.(rest.Creater).Create instead of the setFn?